platform/security/certificate/security-certificate-generator/src/main/java/org/codice/ddf/security/certificate/generator/CertificateAuthority.java

```diff
@@ -58,13 +59,17 @@ public CertificateAuthority(X509Certificate certificate, PrivateKey privateKey)
     }
 
     public KeyStore.PrivateKeyEntry sign(CertificateSigningRequest csr) {
-        //Converters, holders, and builders! Oh my!
-        JcaX509v3CertificateBuilder builder = csr.newCertificateBuilder(getCertificate());
-        X509CertificateHolder holder = builder.build(getContentSigner());
-        JcaX509CertificateConverter converter = newCertConverter();
         X509Certificate signedCert;
+
         try {
+            //Converters, holders, and builders! Oh my!
+            JcaX509v3CertificateBuilder builder = csr.newCertificateBuilder(getCertificate());
+            X509CertificateHolder holder = builder.build(getContentSigner());
+            JcaX509CertificateConverter converter = newCertConverter();
+
             signedCert = converter.getCertificate(holder);
+        } catch (CertIOException e) {
+            throw new CertificateGeneratorException("Could not create signed certificate.", e);
```

Comment:
The second catch deals with the wrapper exception by propagating the cause. I do not believe there is any value here in keeping that wrapper around. As for catching RuntimeException, I any occurs then they should be runtime errors and should be left to bubble out.